c_src/sqlite3_nif.c

```diff
@@ -632,6 +632,73 @@ exqlite_transaction_status(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
       autocommit == 0 ? make_atom(env, "transaction") : make_atom(env, "idle"));
 }
 
+static ERL_NIF_TERM
+exqlite_serialize(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
+{
+    assert(env);
+
+    connection_t* conn = NULL;
+    unsigned char* buffer = NULL;
+    sqlite3_int64 buffer_size = 0;
+    ERL_NIF_TERM serialized;
+
+    if (argc != 1) {
+        return enif_make_badarg(env);
+    }
+
+    if (!enif_get_resource(env, argv[0], connection_type, (void**)&conn)) {
+        return make_error_tuple(env, "invalid_connection");
+    }
+
+    buffer = sqlite3_serialize(conn->db, "main", &buffer_size, 0);
+    if (!buffer) {
+        return make_error_tuple(env, "serialization_failed");
+    }
+
+    serialized = make_binary(env, buffer, buffer_size);
```

Comment:
One alternative to copying the buffer into a BEAM binary here could be to treat the data as a resource and re-use the buffer for a subsequent deserialize call.

This felt a little safer at the expense of some copying.

Reply:
I think this is the safest bet. Otherwise we'd need to spin up some mutex locks to ensure that two writes don't happen to the same buffer.